pub/lib/dependabot/pub/version.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "dependabot/utils"
+require "rubygems_version_patch"
+
+# Dart pre-release versions use 1.0.1-rc1 syntax, which Gem::Version
+# converts into 1.0.1.pre.rc1. We override the `to_s` method to stop that
+# alteration.
+#
+# For details on versions syntax supported by pub, see:
+# https://semver.org/spec/v2.0.0-rc.1.html
+#
+# For details on semantics of version ranges as understood by pub, see:
+# https://github.com/dart-lang/pub_semver
+
+module Dependabot
+  module Pub
+    class Version < Gem::Version
+      VERSION_PATTERN = Gem::Version::VERSION_PATTERN + "(\\+[0-9a-zA-Z\\-.]+)?"
+      ANCHORED_VERSION_PATTERN = /\A\s*(#{VERSION_PATTERN})?\s*\z/.freeze
+
+      attr_reader :build_info
+
+      def initialize(version)
+        @version_string = version.to_s
+        version, @build_info = version.to_s.split("+") if version.to_s.include?("+")
```

Comment:
I don't see this `build_version` referenced anywhere, if we don't need it, maybe we can just drop the attr_reader and defining it?

```suggestion
        version, _ = version.to_s.split("+") if version.to_s.include?("+")
```

Reply:
Yeah, I guess we could throw it away..

Though if instances of `Pub::Version` are every compared or ordered, we ought to consider `build_info`.
I guess we should consider overriding `eql?`: https://docs.ruby-lang.org/en/2.6.0/Gem/Version.html#method-i-eql-3F

Reply:
Yeah that makes sense 👍 I'd say we can always pull it in when we need it, but I don't feel strongly about removing it or keeping it for now, I'll leave it up to y'all

Reply:
I think we'll leave this for now.